hibernate-core/src/test/java/org/hibernate/event/EmbeddableCallbackTest.java

```diff
@@ -67,6 +67,65 @@ public void testNullEmbeddable() {
 		} );
 	}
 
+	@Test
+	@TestForIssue(jiraKey = "HHH-13829")
+	public void testCollectionOfEmbeddable() {
+		AtomicReference<User> user  = new AtomicReference<>(new User());
```

Comment:
I am curious why `AtomicReference` was used. Seems everything is happening in one thread.

Reply:
I use `AtomicReference`  in order to access `user `object both inside and outside lambda expression. Without `AtomicReference` this is not possible.

Reply:
I gained some new knowledge, Thanks. I searched in the testing folder and found quite some precedence. Guess testing class field is also viable, but this seems enjoy better code readability, though my initial doubt would be pretty common.

Reply:
IMHO, org.apache.commons.lang3.mutable.MutableObject (which seems to have been in our class path) seems a better alternative. Sort of misuse of AtomicReference for we are not using it for concurrency reason.

Reply:
MutableObject from apache commons seems more appropriate in this case but does not exist in classpath. Do you think it's better to add it or is it ok with `AtomicReference`?

Reply:
@NathanQingyangXu we are not going to add any apache commons dependencies.

Lambdas in Java often do require the use of `AtomicReference`, `AtomicInteger`, etc.  We run into this alot in 6.0 development where we do use lambdas as part of the code base.  It's an unfortunate necessity with lambdas.

However, @gkatiforis, here you do not really have to do it this way.  You are using `org.hibernate.testing.transaction.TransactionUtil#doInJPA`, but could also use `org.hibernate.testing.transaction.TransactionUtil2#fromTransaction` instead which returns a value.  

Personally I prefer `TransactionUtil2` over `TransactionUtil`.  Yes I wrote it, but the preference is more than that.  I like the naming and structure better in the `TransactionUtil2` contract

Reply:
E.g.,

````
final User user = fromTransaction(
        this::entityManagerFactory,
        em -> {
            final User u = ...;
            ...
            em.persist( u );
            ...
            return u;
        }
);
````

To be fair, this is a test and the performance is not so critical.  But still, good to be conscious of better performance opportunities regardless.  Lambdas in Java have quite a few performance considerations.

Reply:
To my knowledge, Lambdas (and previous inner instance class method) requires 'final' or 'effective final' (the rationale behind the language design is because of concurrency, but that is another story). The reason it is used in this PR is simply because it is a object wrapper. It seems this PR is not using its concurrency feature for everything is running in one thread (I agree if we are involving SessionFactory or other thread-safe object we need them). Also, we only use its 'getter' and 'setter' and any object wrapper (regardless of whether it is thead-safe or not) will work in this specific scenario. That is why I said I think it is sort of misuse.

But for sure, we should use your method. Good to learn of its necessity.

Reply:
`org.hibernate.testing.transaction.TransactionUtil2#fromTransaction` seems ideal for this case, so i  will remove `AtomicReference`. Thank you for pointing this out.